### PR TITLE
Execute Controls before auxiliary system gets evaluated

### PR DESCRIPTION
**Pull request by andrsd**
## Reason

Currently, the control system gets evaluated after the auxiliary system. This prevents us to control the auxiliary system correctly. The values in the aux system are lagged by 1 time step.

## Design

Move the Controls evaluating before aux-system evaluation.

## Impact

Correct behavior when controlling auxiliary system.

### CODE REVIEW

**Comment by andrsd** [comment]
The failure in THM is expected, we know that the failing tests have lagged values.

File: framework/src/problems/FEProblemBase.C [hunk 1]
```diff
@@ -3705,6 +3705,9 @@ FEProblemBase::execute(const ExecFlagType & exec_type)
       _displaced_problem->setCurrentlyComputingJacobian(true);
   }
 
+  if (exec_type != EXEC_INITIAL)
```
Comment (dschwen) [inline framework/src/problems/FEProblemBase.C:3708]: how's the order during INITIAL?

**Comment by lindsayad** [comment]
@dschwen looks like you've got the review handled?

File: framework/src/problems/FEProblemBase.C (hunk 1 above)
Comment (andrsd) [reply framework/src/problems/FEProblemBase.C:3708]: Roughly like so:
```
if (!_app.isRecovering())
{
  ...
  execute(EXEC_INITIAL);
  compute initial material Loop
}
executeControls(EXEC_INITIAL);
```

So, if I understand correctly, `execute` is optional during app recovering. When starting up, there needs to be an extra material loop. So this prevents evaluating the control system twice during `execute_on = initial`

**Comment by andrsd** [comment]
Hi @dschwen . I answered your question about `INITIAL`, if it is satisfactory, please resolve the discussion and potentially approve this MR if you find a few minutes. I would like to get this off my TODO list and finally patch the affected codes...

**Review by dschwen — APPROVED** [review]
Sorry, but yes, this seems fine.

**Comment by dschwen** [comment]
@andrsd has patches ready for the broken apps

**Review by GiudGiud — APPROVED** [review]
good catch!

**Pull request merged**